### PR TITLE
Update terragrunt support to use terragrunt run-all plan / apply / destroy #896

### PR DESCRIPTION
**Pull request by davidtam**
As described in #896 replacing deprecated terragrunt commands with `run-all`

### CODE REVIEW

**Review by brikis98 — APPROVED** [review]
LGTM, thank you for the fix! I'll kick off tests now.

**Comment by brikis98** [comment]
OK, there were two test failures:

* `TestTerraformRedeployExample`. I think this is a transient issue with syslog. Ignore for now, my guess is it'll pass on re-run.
* `TestTgApplyUseLockNoError`. This looks related to this PR. Log output below. Could you look into it?

```
TestTgApplyUseLockNoError 2021-05-25T11:41:51Z logger.go:66: [0mtest = "Hello, World"
=== CONT  TestTgApplyUseLockNoError
    apply_test.go:201: 
        	Error Trace:	apply_test.go:201
        	Error:      	"time=2021-05-25T11:41:46Z level=info msg=Stack at /tmp/TestTgApplyUseLockNoError789882540/terragrunt-no-error:
        	            	  => Module /tmp/TestTgApplyUseLockNoError789882540/terragrunt-no-error (excluded: false, dependencies: [])
        	            	
        	            	[0m[1mInitializing the backend...[0m
        	            	
        	            	[0m[1mInitializing provider plugins...[0m
        	            	
        	            	[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m
        	            	[0m[32m
        	            	You may now begin working with Terraform. Try running "terraform plan" to see
        	            	any changes that are required for your infrastructure. All Terraform commands
        	            	should now work.
        	            	
        	            	If you ever set or change modules or backend configuration for Terraform,
        	            	rerun this command to reinitialize your working directory. If you forget, other
        	            	commands will detect it and remind you to do so if necessary.[0m
        	            	[0m[1m[32m
        	            	Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
        	            	[0m[0m[1m[32m
        	            	Outputs:
        	            	
        	            	[0mtest = "Hello, World"" does not contain "-lock=true"
        	Test:       	TestTgApplyUseLockNoError
```

**Comment by davidtam** [comment]
I made some changes and added debug (see `RunCommandAndGetOutputE`) and I think `-lock=true` is passed down correctly but the output no longer contains that.
I am quite new to tf and tg tbh - is there a flag to get the -lock=true printed in output?
```
=== CONT  TestTgApplyUseLockNoError
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 retry.go:91: terragrunt [run-all apply -input=false -auto-approve -lock=true --terragrunt-non-interactive]
RunCommandAndGetOutputE {terragrunt [run-all apply -input=false -auto-approve -lock=true --terragrunt-non-interactive] /var/folders/ll/xp4ptlrd0cxcmpz0pbh41k3c0000gp/T/TestTgApplyUseLockNoError061216298/terragrunt-no-error map[] <nil>}
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: Running command terragrunt with args [run-all apply -input=false -auto-approve -lock=true --terragrunt-non-interactive]
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: time=2021-05-25T13:50:51+01:00 level=info msg=Stack at /private/var/folders/ll/xp4ptlrd0cxcmpz0pbh41k3c0000gp/T/TestTgApplyUseLockNoError061216298/terragrunt-no-error:
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66:   => Module /private/var/folders/ll/xp4ptlrd0cxcmpz0pbh41k3c0000gp/T/TestTgApplyUseLockNoError061216298/terragrunt-no-error (excluded: false, dependencies: [])
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: Initializing the backend...
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: Initializing provider plugins...
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: Terraform has been successfully initialized!
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: You may now begin working with Terraform. Try running "terraform plan" to see
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: any changes that are required for your infrastructure. All Terraform commands
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: should now work.
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: If you ever set or change modules or backend configuration for Terraform,
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: rerun this command to reinitialize your working directory. If you forget, other
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: commands will detect it and remind you to do so if necessary.
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: Outputs:
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: 
TestTgApplyUseLockNoError 2021-05-25T13:50:51+01:00 logger.go:66: test = "Hello, World"
```

**Comment by davidtam** [comment]
ok I further added `TF_LOG=debug` in the failing test do that `-lock=true` is present in output and hence could be asserted.  hope this is ok?

File: modules/terraform/format.go [hunk 1]
```diff
@@ -14,9 +14,8 @@ import (
 var TerraformCommandsWithLockSupport = []string{
 	"plan",
 	"apply",
-	"apply-all",
+	"run-all",
 	"destroy",
-	"destroy-all",
```
Comment (brikis98) [inline modules/terraform/format.go:19]: For backwards compatibility, we may need to keep these?

File: modules/terraform/apply_test.go [hunk 2]
```diff
@@ -192,6 +192,7 @@ func TestTgApplyUseLockNoError(t *testing.T) {
 		TerraformDir:    testFolder,
 		TerraformBinary: "terragrunt",
 		Lock:            true,
+		EnvVars:         map[string]string{"TF_LOG": "DEBUG"},   // debug level to get -lock CLI option passed down
```
Comment (brikis98) [inline modules/terraform/apply_test.go:195]: Ah, nice 👍

File: modules/terraform/format.go (hunk 1 above)
Comment (davidtam) [reply modules/terraform/format.go]: ok I have added back those...

**Review by brikis98 — DISMISSED** [review]
Great, thanks! I'll kick off tests now.

**Comment by brikis98** [comment]
Pre-commit hook failed:

```
[INFO] Initializing environment for https://github.com/gruntwork-io/pre-commit.
goimports................................................................Failed
- hook id: goimports
- files were modified by this hook

modules/terraform/apply_test.go
```

Please run `goimports`!

**Comment by davidtam** [comment]
I am really sorry - I have run goimports and push again...

**Comment by brikis98** [comment]
No prob! I'll re-run tests now.

**Review by brikis98 — APPROVED** [review]
Tests passed! Merging now.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.35.0

**Comment by davidtam** [comment]
great!! thank you!!

**Pull request merged**